language/move-vm/types/src/values/values_impl.rs

```diff
@@ -245,15 +245,23 @@ impl ValueImpl {
 }
 
 impl Value {
-    pub fn is_valid_script_arg(&self, sig: &SignatureToken) -> bool {
+    // REVIEW: this allows primitive (including `Sender`) and vectors of them. Vectors can only
+    // be one level (a different organization of `ValueImpl` could give us that property in an
+    // efficient way). Conceptually if we serialize the value and deserialize according to
+    // the argument type we would have accomplished a proper check
+    pub fn is_valid_arg(&self, sig: &SignatureToken) -> bool {
```

Comment:
this is an interpreter entry point that needs some work/love from @vgao1996.
In principle here we would like to accept anything as long as it matches the type but that may be a difficult story.
Also changes in `ValueImpl` may affect this so @vgao1996 please take a look and think about it

Reply:
Yeah as you mentioned offline, we should come up with a function checking if a `Value` matches its type here. I'll take over this part once I'm back to work on the values. (It should be easier with the cleanup in the data cache.)